src/plugins/discover/public/application/main/utils/update_saved_search.ts

```diff
@@ -48,11 +48,20 @@ export function updateSavedSearch({
       .setField('filter', services.data.query.filterManager.getFilters());
   } else if (state) {
     const appFilters = state.filters ? cloneDeep(state.filters) : [];
-    const globalFilters = globalStateContainer.get()?.filters ?? [];
+    const globalFilters = cloneDeep(globalStateContainer.get()?.filters ?? []);
+    // remove filters which are also present in appState
+    const cleanedGlobalFilters = globalFilters.filter((globalFilter) => {
+      const filterInAppState = appFilters.find(
+        (appFilter) =>
+          isEqual(appFilter.meta, globalFilter.meta) && isEqual(appFilter.query, globalFilter.query)
+      );
+
+      return !filterInAppState;
+    });
 
     savedSearch.searchSource
       .setField('query', state.query ?? undefined)
-      .setField('filter', [...appFilters, ...globalFilters]);
+      .setField('filter', [...appFilters, ...cleanedGlobalFilters]);
```

Comment:
This seems to be fixing internal state but it should also sync to filter bar which does not happen automatically yet.